packages/graphql/src/schema/augment/fulltext.ts

```diff
@@ -17,50 +17,114 @@
  * limitations under the License.
  */
 
-import { GraphQLFloat, GraphQLNonNull, GraphQLString } from "graphql";
+import type { GraphQLEnumValueConfigMap } from "graphql";
+import { GraphQLEnumType, GraphQLFloat, GraphQLNonNull, GraphQLString } from "graphql";
 import type { SchemaComposer } from "graphql-compose";
 import type { Node } from "../../classes";
 import { fulltextResolver } from "../resolvers/query/fulltext";
 import { upperFirst } from "../../utils/upper-first";
 import { FloatWhere } from "../../graphql/input-objects/FloatWhere";
 import { SCORE_FIELD } from "../../graphql/directives/fulltext";
 
-export const fulltextArgDeprecationMessage =
+export function createFulltextDeprecationMessage(node: Node, type = "argument") {
+    return `This ${type} has been deprecated and will be removed in version 4.0.0 of the library.
+Please use the fulltext argument in ${node.plural}Connection with index you wish to query instead.
+More information about the changes to @fulltext can be found here:
+https://neo4j.com/docs/graphql-manual/current/guides/v4-migration/#_fulltext_changes.
+`;
+}
+
+const fulltextArgDeprecationMessageInArgument =
     "This argument has been deprecated and will be removed in version 4.0.0 of the library. " +
-    "Please use the top-level query that corresponds to the index you wish to query instead. " +
+    "Please use the phrase and index arguments instead. " +
     "More information about the changes to @fulltext can be found here: " +
     "https://neo4j.com/docs/graphql-manual/current/guides/v4-migration/#_fulltext_changes.";
 
+function createEnumType(name: string, rawValues: string[]): GraphQLEnumType {
+    const values = rawValues.reduce((acc, value) => {
+        return {
+            ...acc,
+            [value]: {
+                value,
+            },
+        };
+    }, {} as GraphQLEnumValueConfigMap);
+
+    return new GraphQLEnumType({
+        name: `${name}FulltextIndexes`,
+        values,
+    });
+}
+
 export function augmentFulltextSchema(
     node: Node,
     composer: SchemaComposer,
     nodeWhereTypeName: string,
     nodeSortTypeName: string
 ) {
     if (node.fulltextDirective) {
-        const fields = node.fulltextDirective.indexes.reduce((res, index) => {
+        const indexes = node.fulltextDirective.indexes.map((index) => {
+            const indexName = index.indexName || index.name;
+            if (indexName === undefined) {
+                throw new Error("The name of the fulltext index should be defined using the indexName argument.");
```

Comment:
Would it make sense to throw a `Neo4jGraphQLError` here?

Reply:
Maybe? We don't have a standard for errors sadly. This one is more of a fatal error I think, as the graphql typings should avoid this case, so I'm not sure if `Neo4jGraphQLError` is the appropriate type, but happy to change it if you think is better